e2e/tests/interchain_accounts/base_test.go

```diff
@@ -228,3 +238,177 @@ func (s *InterchainAccountsTestSuite) TestMsgSendTx_FailedTransfer_InsufficientF
+			banktypes.RegisterInterfaces(cfg.InterfaceRegistry)
+			cdc := codec.NewProtoCodec(cfg.InterfaceRegistry)
+
+			bz, err := icatypes.SerializeCosmosTx(cdc, []proto.Message{msgSend})
+			s.Require().NoError(err)
+
+			packetData := icatypes.InterchainAccountPacketData{
+				Type: icatypes.EXECUTE_TX,
+				Data: bz,
+				Memo: "e2e",
+			}
+
+			msgSendTx := controllertypes.NewMsgSendTx(controllerAccount.Bech32Address(chainA.Config().Bech32Prefix), ibctesting.FirstConnectionID, uint64(1), packetData)
+
+			resp, err := s.BroadcastMessages(
+				ctx,
+				chainA,
+				controllerAccount,
+				msgSendTx,
+			)
+
+			s.AssertValidTxResponse(resp)
+			s.Require().NoError(err)
+
+			time.Sleep(1 * time.Second)
+		})
+	})
+
+	t.Run("start relayer", func(t *testing.T) {
+		s.StartRelayer(relayer)
+	})
+
+	t.Run("verify channel is closed due to timeout on ordered channel", func(t *testing.T) {
+		channel, err := s.QueryChannel(ctx, chainA, portID, "channel-1")
+		s.Require().NoError(err)
+
+		t.Logf("channel-1: %+v", channel)
+		s.Require().Equal(channeltypes.CLOSED, channel.State)
+	})
+
+	t.Run("verify tokens not transferred", func(t *testing.T) {
+		balance, err := chainB.GetBalance(ctx, chainBAccount.Bech32Address(chainB.Config().Bech32Prefix), chainB.Config().Denom)
+		s.Require().NoError(err)
+
+		_, err = chainB.GetBalance(ctx, hostAccount, chainB.Config().Denom)
+		s.Require().NoError(err)
+
+		expected := testvalues.StartingTokenAmount
+		s.Require().Equal(expected, balance)
+	})
+
+	// re-register interchain account to reopen the channel now that it has been closed due to timeout
+	// on an ordered channel
+	t.Run("register interchain account", func(t *testing.T) {
+		version := getICAVersion(testconfig.GetChainATag(), testconfig.GetChainBTag())
+		msgRegisterInterchainAccount := controllertypes.NewMsgRegisterInterchainAccount(ibctesting.FirstConnectionID, controllerAccount.Bech32Address(chainA.Config().Bech32Prefix), version)
+		err := s.RegisterInterchainAccount(ctx, chainA, controllerAccount, msgRegisterInterchainAccount)
+		s.Require().NoError(err)
+
+		test.WaitForBlocks(ctx, 3, chainA, chainB)
+	})
+
+	t.Run("verify new channel is now open and interchain account has been reregistered with the same portID", func(t *testing.T) {
+		channel, err := s.QueryChannel(ctx, chainA, portID, "channel-2")
+		s.Require().NoError(err)
+
+		t.Logf("channel-2: %+v", channel)
+		s.Require().Equal(channeltypes.OPEN, channel.State)
+	})
+
+	t.Run("broadcast MsgSendTx", func(t *testing.T) {
+		// assemble bank transfer message from host account to user account on host chain
+		msgSend := &banktypes.MsgSend{
+			FromAddress: hostAccount,
+			ToAddress:   chainBAccount.Bech32Address(chainB.Config().Bech32Prefix),
+			Amount:      sdk.NewCoins(testvalues.DefaultTransferAmount(chainB.Config().Denom)),
+		}
+
+		cfg := simappparams.MakeTestEncodingConfig()
+		banktypes.RegisterInterfaces(cfg.InterfaceRegistry)
+		cdc := codec.NewProtoCodec(cfg.InterfaceRegistry)
```

Comment:
We can use [this function](https://github.com/cosmos/ibc-go/blob/48321ffc8fd558f99b9d66f44913f7563364aeed/e2e/testsuite/codec.go#L11) to fetch the codec.